client/my-sites/stats/stats-purchase/use-available-upgrade-tiers.tsx

```diff
@@ -161,7 +143,41 @@ function useAvailableUpgradeTiers(
 		return MOCK_PLAN_DATA;
 	}
 
-	let tiersForUi = commercialProduct.price_tier_list.map( transformTier );
+	const currentTierPrice = usageData.current_tier.minimum_price;
+
+	let tiersForUi = commercialProduct.price_tier_list.map(
+		( tier: PriceTierListItemProps ): StatsPlanTierUI => {
```

Comment:
I would still prefer a dedicated function here for readability; however it's challenging to find a proper name 😄

Reply:
I will refactor it by passing the `currentTierPrice` to a closure function accompanying the callback function.